### DISCUSSION TITLE
Moving a note should update image links

### DISCUSSION THREAD

**Discussion by madeindjs** [GitJournal/GitJournal#805] [Feature Requests]
When you insert a picture on a note placed on a root folder, image are save on root with link like this: `./picture.png`.

Whene you update note's folder, picture not follow new folder and link is not updated so the link become broken.

**Comment by aisbergde** [comment]
That's why I would not use (sub) folders in my notes system but only tags and filenames to distinguish.

To solve this issue a monitoring should be done when moving a file and I should be asked if I want to update the links. Some WIKI pages systems are checking relations when moving files. 

But I would not like that some auto update will change the notes not asking me.

And there is another issue: when moving notes files outside GitJournal directly then no detection could happen and the links will be wrong anyway. So I would not like the developer to spent much energy on change detection, it is not possible to cover all cases :-)

A possible way I see is to use Placeholders or variables. There is an similar issue in SchemaSpy with markdown comments in database documentation because the place of the generated file containing links to images can be on different places:
https://github.com/schemaspy/schemaspy/issues/705

Something like {RootFolder}/images/picture.png could solve the issue.

But this would be a GitJounal specific solution. And the goal should be that the markdown notes are working not only inside GitJournal, but also outside.

I don't know if there are any markdown extensions allowing to define this {RootFolder} as variable. But what I learned, there are so many different markdown versions and the are even not compatible on such "simple" things like lists:

https://github.com/yzhang-gh/vscode-markdown/issues/680#issuecomment-626420198

[Try this. It was a shock for me seeing this very different renderings results of the same markdown in different dialects](https://johnmacfarlane.net/babelmark2/?normalize=1&text=*+A%0A+*+B%0A++*+C%0A+++*+D%0A++++*+E%0A+++++*+F%0A++++++*+G)

That's why I try:
- not to use sub folders for the "main notes system" but to keep all notes in one folder, even if the app supports folders
- if I use folders, then they should be on the same level like
  - notes
  - project1
  - images
When I link from both folders (notes, project1) to the images folder, I could move notes and the link to images will still work and the markdown file content don't needs to be changed.

I try to use markdown elements which are supported by different markdown dialects: My blog page uses one dialect to render (https://github.com/michelf/php-markdown), GitHub and AzureDevOps using other dialects, ...

**Comment by mipro98** [comment]
Trying to automate fixing links in a collection of markdown files is always more of a challenge than one might think. On the one side, most people do not want to break things just by renaming or moving a file but otherwise one doesn't want to have own content automatically manipulated from scripts which always is a risk of unnoticed data loss.

Because of this reason I would also be okay without the App trying to preserve correct links.

**OT:**
Out of frustration on this whole topic I created a python script some time ago to at least detect non-working markdown links in large markdown-files collections. Then, I implemented basic functionality for fixing them if only the path (but not the filename) was wrong. It's basically just some regex magic to detect markdown links and some relative-path conversions. It's not complete and not really tested to a larger extent but maybe it can help someone. It's intended to be used as a python module: https://gist.github.com/mipro98/efca63be2c3c38b70b5ced63df7c648f
A little demo on how one might use this script is here: https://gist.github.com/mipro98/149f18fe569dad7bc385d89c2065e48c
